source/extensions/common/wasm/wasm.cc

```diff
@@ -1571,50 +1565,62 @@ bool Wasm::initialize(const std::string& code, bool allow_precompiled) {
     return false;
   }
 
-  // Construct a unique identifier for the VM based on the provided vm_id and a hash of the
-  // code.
-  vm_id_with_hash_ = vm_id_ + ":" + base64Sha256(code);
+  if (started_from_ == Cloneable::NotCloneable) {
```

Comment:
This is just a bunch of extra `if (started_from_ ...)` guards and indents, but GitHub somehow didn't catch that, so it's best to review locally with `git show -w`.